homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,909 @@
+
+    _LOGGER.debug("Adding device: %s", device)
+    add_devices([device])
+
+
+@asyncio.coroutine
+def fetch_headers(hass, url, headers):
+    """Fetch headers from URL, first by trying HEAD, then by trying a GET."""
+    # try a HEAD request to the source
+    src_response = None
+    try:
+        session = async_get_clientsession(hass)
+        src_response = yield from session.head(url, headers=headers)
+        yield from src_response.release()
+    except aiohttp.ClientError:
+        pass
+
+    if src_response and 200 <= src_response.status < 300:
+        return src_response.headers
+
+    # try a GET request to the source, but ignore all the data
+    session = async_get_clientsession(hass)
+    src_response = yield from session.get(url, headers=headers)
+    yield from src_response.release()
+
+    return src_response.headers
+
+
+class UpnpNotifyView(HomeAssistantView):
+    """Callback view for UPnP NOTIFY messages."""
+
+    url = '/api/dlna_dmr.notify'
+    name = 'api:dlna_dmr:notify'
+    requires_auth = False
+
+    def __init__(self, hass):
+        """Initializer."""
+        self.hass = hass
+        self._registered_services = {}
+        self._backlog = {}
+
+    def register(self, router):
+        """Register the view with a router."""
+        handler = request_handler_factory(self, self.async_notify)
+        router.add_route('notify', UpnpNotifyView.url, handler)
+
+    @asyncio.coroutine
+    def async_notify(self, request):
+        """Callback method for NOTIFY requests."""
+        if 'SID' not in request.headers:
+            return aiohttp.web.Response(status=422)
+
+        headers = request.headers
+        sid = headers['SID']
+        body = yield from request.text()
+
+        # find UpnpService by SID
+        if sid not in self._registered_services:
+            self._backlog[sid] = {'headers': headers, 'body': body}
+            return aiohttp.web.Response(status=202)
+
+        service = self._registered_services[sid]
+        service.on_notify(headers, body)
+        return aiohttp.web.Response(status=200)
+
+    @property
+    def callback_url(self):
+        """Full URL to be called by device/service."""
+        base_url = self.hass.config.api.base_url
+        return urllib.parse.urljoin(base_url, self.url)
+
+    def register_service(self, sid, service):
+        """Register a UpnpService under SID."""
+        if sid in self._registered_services:
+            raise RuntimeError('SID {} already registered.'.format(sid))
+
+        self._registered_services[sid] = service
+
+        if sid in self._backlog:
+            item = self._backlog[sid]
+            service.on_notify(item['headers'], item['body'])
+            del self._backlog[sid]
+
+    def unregister_service(self, sid):
+        """Unregister service by SID."""
+        if sid in self._registered_services:
+            del self._registered_services[sid]
+
+
+class PickyDeviceProxyView(HomeAssistantView):
```

Comment:
I understand the "PickyDevice" issue, but we should be able to hide this issue from HomeAssistant and move its handling to the supporting library. 

I understand that it is no easy task, but today it's Samsung, tomorrow some 10 other different brands, each picky on something different.

Reply:
Fair enough, I'll remove it.

But, people will try to use TTS services and report their devices not playing.
Not sure how to move this to another library. This requires starting a new webserver, mixing it with home assistants event loop, if possible? Is there a way to disable the TTS service for a specific device, from within the component?